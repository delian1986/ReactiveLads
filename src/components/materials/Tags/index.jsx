import { Component } from "react";
import { TagFilter } from "../TagFilter";

export class Tags extends Component {
  componentDidMount() {
    this.props.fetchTags();
  }

  render() {
    return this.props.tags.length > 0 ? (
      <div>
        <h5>Tags</h5>
        {this.props.tags.map((tag) => (
          <TagFilter
            key={tag.id}
            id={tag.id}
            name={tag.name}
            selectTag={this.props.selectTag}
          />
        ))}
      </div>
    ) : (
      <div className="spinner-border" role="status">
        <span className="sr-only">Loading...</span>
      </div>
    );
  }
}
