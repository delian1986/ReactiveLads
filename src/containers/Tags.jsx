import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { Tags } from "../components/materials/Tags";

import { selectTag } from "../actions/filters";

const mapStateToProps = (state) => {
  return {
    tags: state.tags
  };
};

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      selectTag: selectTag
    },
    dispatch
  );

export default connect(mapStateToProps, mapDispatchToProps)(Tags);
