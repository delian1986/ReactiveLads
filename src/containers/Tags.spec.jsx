import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Tags from "./Tags";

const mockStore = configureStore([]);

describe("tests <Tags/> component", () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      tags: [{ id: 1, name: "data" }],
      filters: {
        selectedTags: [3, 5]
      }
    });
    store.dispatch = jest.fn();
    component = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Tags component={() => <></>} />
        </Provider>
      </BrowserRouter>
    );
  });
  describe("Test rendering component", () => {
    it("should render with given state from Redux store", () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
  describe("Actions", () => {
    it("should dispatch an action on button click", () => {
      renderer.act(() => {
        component.root.findByType("input").props.onChange();
      });

      expect(store.dispatch).toHaveBeenCalledTimes(1);
    });
  });
});
