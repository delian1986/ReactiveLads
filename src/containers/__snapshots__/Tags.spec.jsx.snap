// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests <Tags/> component Test rendering component should render with given state from Redux store 1`] = `
Array [
  <h5>
    Tags
  </h5>,
  <div>
    <div>
      <label>
        <input
          checked={false}
          className="mr-2"
          onChange={[Function]}
          type="checkbox"
        />
        data
      </label>
    </div>
  </div>,
]
`;
